### PR TITLE
Profile icon in tab bar

### LINKED ISSUE #1165: [2.0] Replace Profile tab with Instance tab when logged in as guest

### LINKED ISSUE #1166: [2.0] Show avatar of logged-in user in tab bar

### PR DESCRIPTION
**Pull request by Sjmarf**
Shows the profile picture in the tab bar and displays an `InstanceView` in the profile tab when using a guest account.

Closes #1165, closes #1166

### CODE REVIEW

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Looks good!

Profile tab images appear to get a little squished in the tab bar, though--this picture should show a perfect square, and does in the profile view, but in the tab bar it's a rectangle:

Screenshot 2024-08-02 at 10 36 29 PM

Screenshot 2024-08-02 at 10 39 22 PM

**Comment by Sjmarf** [comment]
Nice catch, fixed 👍

File: Mlem/App/Views/Root/ContentView.swift [hunk 1]
```diff
@@ -131,4 +145,34 @@ struct ContentView: View {
...
+                selectedAvatarImage = image
+                    .resized(to: .init(width: image.size.width / image.size.height * 26, height: 26))
+                    .circleBorder(color: .init(palette.accent), width: 3.5)
+                    .withRenderingMode(.alwaysOriginal)
+            }
+            
+            Task { @MainActor in
+                self.avatarImage = avatarImage
+                self.selectedAvatarImage = selectedAvatarImage
+            }
```
Comment (EricBAndrews) [inline Mlem/App/Views/Root/ContentView.swift:165]: This block causes build failures in Xcode 15:

`Reference to captured var 'avatarImage' in concurrently-executing code`
`Reference to captured var 'selectedAvatarImage' in concurrently-executing code`

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Looks good!

There's a build failure against Xcode 15, though--see comment below.

File: Mlem/App/Views/Root/ContentView.swift (hunk 1 above)
Comment (Sjmarf) [reply Mlem/App/Views/Root/ContentView.swift:165]: Fixed 👍

**Review by EricBAndrews — APPROVED** [review]
LGTM! 🔥 

There are a couple build issues still present due to the MlemMiddleware version, but they should all be resolved by the dev2 merge.

**Pull request merged**